stablehlo/compatibility/dialect/VersionedStablehloBase.td

```diff
@@ -0,0 +1,182 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2022 The VersionedHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_COMPATIBILITY_VERSIONEDHLO_BASE
+#define STABLEHLO_COMPATIBILITY_VERSIONEDHLO_BASE
+
+include "mlir/IR/OpBase.td"
+
+//===----------------------------------------------------------------------===//
+// Versioned Dialect Constants and Declaration
+//===----------------------------------------------------------------------===//
+
+def VersionedHLO_Dialect : Dialect {
+  let name = "versionedhlo";
+  let cppNamespace = "::mlir::versionedhlo";
+
+  let description = [{
+    A shim opset of Versioned StableHLO ops for versions 0.x.x and 1.x.x.
+    
+    Version log:
+      0.0.0: Base release.
+      0.1.0: CustomCallOp add result aliasing.
```

Comment:
I didn't know internally we have a name for the base release. The proposal in the Compat RFC is that all changes to the VHLO opset are accompanied by a minor version bump, since they have backward/forward compatibility implications, more than a patch. Patch bumps would be reserved for compat-exempt fixes to documentation or verifiers.

Also I agree that we should do this in a followup ticket. We could discuss batching there as well:

```
    Version log:
      0.3.0: Bootstrap from MHLO: https://github.com/openxla/stablehlo/pull/1.
      0.4.0: Add AllGatherOp::use_global_device_ids: https://github.com/openxla/stablehlo/pull/272.
             Add CollectivePermuteOp::channel_id: https://github.com/openxla/stablehlo/pull/388.
             Add CustomCallOp::output_operand_aliases: https://github.com/openxla/stablehlo/pull/403.
```

Reply:
I modified this to include PR links, but am going to wait to call the base release 0.3.0 until we align on what these numbers mean. I'm not sure the internal numbering scheme aligns with the external release numbering proposed in the Compat RFC.

Reply:
Sounds good, will update to use this numbering scheme.